csrc/pf_inner.c

```diff
@@ -1106,6 +1106,15 @@ DBUG(("XX ah,m,l = 0x%8x,%8x,%8x - qh,l = 0x%8x,%8x\n", ah,am,al, qh,ql ));
 	    }
 	    endcase;
 
+	case ID_FILE_RESIZE: /* ( ud fileid -- ior ) */
+	    {
+		FileStream *File = (FileStream *) TOS;
+		ucell_t SizeHi = (ucell_t) M_POP;
+		ucell_t SizeLo = (ucell_t) M_POP;
+		TOS = sdResizeFile( File, SizeLo, SizeHi );
```

Comment:
Indeed. I would use uint64_t instead of off_t, though.  That's in C99 and it seems unlikely that somebody wants to work with files larger than 2^64 bytes.